File: server/ruby/app/infrastructure/mappers/payment_details/reattempt_instructions_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class ReattemptInstructionsMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::ReattemptInstructions.new(
            conditions: Infrastructure::Mappers::PaymentDetails::ReattemptInstructionsConditionsMapper.from_sdk_response(response.conditions),
            frozen_period: response.frozen_period,
            indicator: response.indicator
          )
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/reattempt_instructions_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.reattempt_instructions import ReattemptInstructions
from onlinepayments.sdk.domain.reattempt_instructions import ReattemptInstructions as ReattemptInstructionsSdk
from app.infrastructure.mappers.payment_details.reattempt_instructions_conditions_mapper import ReattemptInstructionsConditionsMapper

class ReattemptInstructionsMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[ReattemptInstructionsSdk]) -> Optional[ReattemptInstructions]:
        if response is None:
            return None

        dto = ReattemptInstructions()
        dto.conditions = ReattemptInstructionsConditionsMapper.map_from_sdk_response(response.conditions)
        dto.frozen_period = response.frozen_period
        dto.indicator = response.indicator

        return dto